IdentityCore/src/workplacejoin/ios/MSIDWorkPlaceJoinUtil.m

```diff
@@ -29,140 +29,108 @@
         // We never should hit this error anyways, as any of this stuff being missing will cause failures farther up.
         if (error)
         {
-            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Missing some piece of WPJ data", nil, nil, nil, context.correlationId, nil);
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Missing some pieces of WPJ data", nil, nil, nil, context.correlationId, nil);
         }
         
         return nil;
     }
-    
-    {
-        MSIDRegistrationInformation *info = [[MSIDRegistrationInformation alloc] initWithSecurityIdentity:identity
-                                                                                        certificateIssuer:certificateIssuer
-                                                                                              certificate:certificate
-                                                                                       certificateSubject:certificateSubject
-                                                                                          certificateData:certificateData
-                                                                                               privateKey:privateKey];
-        return info;
-    }
-_error:
-    if (identity)
+    else
     {
-        CFRelease(identity);
+        info = [[MSIDRegistrationInformation alloc] initWithSecurityIdentity:identity
+                                                           certificateIssuer:certificateIssuer
+                                                                 certificate:certificate
+                                                          certificateSubject:certificateSubject
+                                                             certificateData:certificateData
+                                                                  privateKey:privateKey];
+        
     }
-    if (certificate)
+    
+    CFReleaseNull(identity);
+    CFReleaseNull(certificate);
+    CFReleaseNull(privateKey);
+    
+    return info;
+}
+
++ (SecIdentityRef)copyWPJIdentity:(id<MSIDRequestContext>)context
+                sharedAccessGroup:(NSString *)accessGroup
+                certificateIssuer:(NSString **)issuer
+
+{
+    NSMutableDictionary *identityDict = [[NSMutableDictionary alloc] init];
+    [identityDict setObject:(__bridge id)kSecClassIdentity forKey:(__bridge id)kSecClass];
+    [identityDict setObject:(__bridge id)kCFBooleanTrue forKey:(__bridge id)kSecReturnRef];
+    [identityDict setObject:(__bridge id) kSecAttrKeyClassPrivate forKey:(__bridge id)kSecAttrKeyClass];
+    [identityDict setObject:(__bridge id)kCFBooleanTrue forKey:(__bridge id)kSecReturnAttributes];
+    [identityDict setObject:accessGroup forKey:(__bridge id)kSecAttrAccessGroup];
+    
+    CFDictionaryRef result = NULL;
+    OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)identityDict, (CFTypeRef *)&result);
+    
+    if (status != errSecSuccess)
```

Comment:
Do we want to pass and log the error?

Reply:
As discussed offline, not finding a WPJ identity is not an error condition. The caller of the function copyWPJIdentity
will log @"Failed to retrieve WPJ identity." incase we fail to retrieve identity. We will log error in case pieces of the WPJ identity like the certificate. issuer, e.t.c is missing.